libopenage/engine.cpp

```diff
@@ -548,4 +542,12 @@ void Engine::move_phys_camera(float x, float y, float amount) {
 	this->engine_coord_data->camgame_phys += cam_delta.to_phys3();
 }
 
+void Engine::start_game(std::unique_ptr<GameMain> game) {
+	// TODO: maybe implement a proper 1-to-1 connection
+	assert(game);
```

Comment:
Asserts are removed by the preprocessor in release, so there is a need for additional macros that is mimicking or using NDEBUG, if we go that way.

Something like:
```cpp
#ifdef EXCEPTION_ASSERT
	#ifndef NDEBUG
		#include <exception>
		#define Assert(expr) \
			((expr) \
				? static_cast<void> (0) \
				: throw std::exception(#expr##__FILE__##__LINE__##__func__)))
	#else
		#define Assert(expr) (void)0
	#endif
#else
	#include <cassert>
	#define Assert(expr) assert(expr)
#endif
```